# System setup - checking GPU and Torch

First we want to see if our system can detect a GPU:

In [1]:
!nvidia-smi

Sat Jul  4 20:54:56 2026       
+---------------------------------------------------------------------------------------+
| NVIDIA-SMI 535.183.01             Driver Version: 535.183.01   CUDA Version: 12.2     |
|-----------------------------------------+----------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |         Memory-Usage | GPU-Util  Compute M. |
|                                         |                      |               MIG M. |
|=========================================+======================+======================|
|   0  NVIDIA GeForce RTX 3060 ...    Off | 00000000:01:00.0  On |                  N/A |
| N/A   39C    P5              17W /  80W |   1623MiB /  6144MiB |     41%      Default |
|                                         |                      |                  N/A |
+-----------------------------------------+----------------------+--

This output will vary system to system, but for mine:
- NVIDIA driver is installed (535.183.01)
- A GPU is detected (RTX 3060) 👴
- A CUDA runtime is available (12.2)
- The GPU is currently in use (17W / 1.6 Gb)

We still need to know if Pytorch can find the GPU:

In [2]:
import torch

In [5]:
if torch.cuda.is_available():
    print("PyTorch version:", torch.__version__)
    print("CUDA available:", torch.cuda.is_available())
    print("CUDA version (PyTorch build):", torch.version.cuda)
    print("GPU count:", torch.cuda.device_count())
    
    for i in range(torch.cuda.device_count()):
        print(f"GPU name: {i+1}: {torch.cuda.get_device_name(i)}")


PyTorch version: 2.7.1+cu128
CUDA available: True
CUDA version (PyTorch build): 12.8
GPU count: 1
GPU name: 1: NVIDIA GeForce RTX 3060 Laptop GPU


Fantastic! Everything we need for YOLO training is available.